src/Examine.Core/Search/IFaceting.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Examine.Search
+{
+    /// <summary>
+    /// Faceting operations
+    /// </summary>
+    public interface IFaceting : IQueryExecutor
+    {
+        /// <summary>
+        /// Add a facet string to the current query
+        /// </summary>
+        /// <param name="field"></param>
+        /// <returns></returns>
+        IFacetQueryField WithFacet(string field);
```

Comment:
From your previous comments about removing the And() (IFacetAppending), this could be changed to be:

```cs
IFaceting WithFacet(string field, Action<IFacetQueryField> field = null);
```

Then you can do:

```cs
.WithFacets(facets => 
   facets
       .Facet("nodeName", field => field.MaxCount(10).FieldName("nodeNameFacet"))
       .Facet("test", field => field.MaxCount(10)))
```

Then there is no need for IFacetAppending and the fluent API is a little more succinct since it's no longer combining MaxCount, FieldName and And at the same level. 

What do you think?

Reply:
Great!